### PR TITLE
sys-libs/compiler-rt-sanitizers: Filter -D_FORTIFY_SOURCE away

### PR DESCRIPTION
**Pull request by lanodan**
Closes: https://bugs.gentoo.org/636604
Signed-off-by: Haelwenn (lanodan) Monnier <contact@hacktivis.me>

### CODE REVIEW

**Comment by lanodan** [comment]
I'm also modifying stabilized ebuilds there, not sure if that's allowed, could remove those.

File: sys-libs/compiler-rt-sanitizers/compiler-rt-sanitizers-13.0.0.9999.ebuild [hunk 1]
```diff
@@ -102,6 +102,7 @@ src_configure() {
 		local -x CC=${CHOST}-clang
 		local -x CXX=${CHOST}-clang++
 		strip-unsupported-flags
+		filter-flags '-D_FORTIFY_SOURCE*'
```
Comment (mgorny) [inline sys-libs/compiler-rt-sanitizers/compiler-rt-sanitizers-13.0.0.9999.ebuild:105]: Don't you need it for gcc too, though?

Comment (lanodan) [reply sys-libs/compiler-rt-sanitizers/compiler-rt-sanitizers-13.0.0.9999.ebuild]: After testing that, true, I had that modification for a while, I guess I got confused by `strip-unsupported-flags`.

Fixed it.

**Comment by mgorny** [comment]
@gentoo/toolchain is this a good idea? Sounds a bit like un-hardening to me but it might be necessary anyway for sanitizers.

**Comment by lanodan** [comment]
Any updates on this topic?

**Comment by mgorny** [comment]
```
<@dilfridge> https://github.com/google/sanitizers/issues/247
<@dilfridge> https://reviews.llvm.org/D92521
<@dilfridge> https://reviews.llvm.org/D93877
<@dilfridge> ^ the last one is what you want to backport
```

Could you try that patch instead?

**Comment by lanodan** [comment]
> Could you try that patch instead?

Tried it against 11.1.0 and it works fine. Will change & test the other ebuilds.

**Comment by mgorny** [comment]
Thanks for the testing. I'm going to add the patch into the patchset, test and then bump them.

**Pull request closed without merging**